src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsDpiSettingUpdater.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.XPath;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    [ApplicableComponents(ProjectComponents.WinForms)]
+    public class WinformsDpiSettingUpdater : IUpdater<IProject>
+    {
+        private const int BufferSize = 65536;
+        private const string HighDPIConfiguration = "/configuration/System.Windows.Forms.ApplicationConfigurationSection";
+
+        private readonly WindowsUtilities _utilities = new();
+        private readonly ILogger<WinformsDpiSettingUpdater> _logger;
+        private ProgramFileSpec _programFileSpec = new();
+
+        public string Id => typeof(WinformsDpiSettingUpdater).FullName;
+
+        public string Title => "Winforms Source Updater";
+
+        public string Description => "Update Winforms Program.cs with HighDpiSetting";
+
+        public BuildBreakRisk Risk => BuildBreakRisk.Low;
+
+        public WinformsDpiSettingUpdater(ILogger<WinformsDpiSettingUpdater> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        private string ProcessAppConfigFile(IProject project)
+        {
+            var hdpiValue = "SystemAware";
+            var appConfigFilePath = project.FindFiles("App.config").FirstOrDefault();
+            if (File.Exists(appConfigFilePath))
+            {
+                var appConfigFile = new ConfigFile(appConfigFilePath);
+                var highDPISetting = appConfigFile.Contents.XPathSelectElement(HighDPIConfiguration);
```

Comment:
It would be nice to abstract the retrieval of keys so that you can just ask for them rather than using xpath...